File: gems/decomplex/examples/python/fat-union.py
def handle(node):
    match node:
        case AST.Call:
            node.line(); node.col(); node.ty(); node.span(); node.parent(); node.recv()
        case AST.Func:
            node.line(); node.col(); node.ty(); node.span(); node.parent(); node.name()
        case AST.Lit:
            node.line(); node.col(); node.ty(); node.span(); node.parent(); node.value()

File: gems/decomplex/examples/rust/fat-union.rs
fn handle(node: Ast) {
    match node {
        AST::Call => {
            node.line();
            node.col();
            node.ty();
            node.span();
            node.parent();
            node.recv();
        }
        AST::Func => {
            node.line();
            node.col();
            node.ty();
            node.span();
            node.parent();
            node.name();
        }
        AST::Lit => {
            node.line();
            node.col();
            node.ty();
            node.span();
            node.parent();
            node.value();
        }
    }
}
